src/VisualStudio/Core/Test.Next/Services/ServiceHubServicesTests.cs

```diff
@@ -458,11 +454,9 @@ private static (Project, Document) GetProjectAndDocument(Solution solution, stri
 
         private async Task UpdatePrimaryWorkspace(RemoteHostClient client, Solution solution)
         {
-            await client.RunRemoteAsync(
-                WellKnownServiceHubService.RemoteHost,
-                nameof(IRemoteHostService.SynchronizePrimaryWorkspaceAsync),
+            await client.TryInvokeAsync<IRemoteAssetSynchronizationService>(
                 solution,
-                new object[] { await solution.State.GetChecksumAsync(CancellationToken.None), _solutionVersion++ },
+                async (service, solutionInfo, cancellationToken) => await service.SynchronizePrimaryWorkspaceAsync(solutionInfo, await solution.State.GetChecksumAsync(cancellationToken), _solutionVersion++, cancellationToken),
```

Comment:
>await solution.State.GetChecksumAsync(cancellationToken) [](start = 129, length = 56)

I'd move `await solution.State.GetChecksumAsync(cancellationToken)` outside the lamda to simplify the expression.

Reply:
➡️ Done